test/stdlib/DispatchData.swift

```diff
@@ -0,0 +1,376 @@
+
+	// Copy from offset 9, but only 1 byte. Ends before the end of the data
+	destPtr[1] = 0xFF
+	destPtr[2] = 0xFF
+	destPtr[3] = 0xFF
+	destPtr[4] = 0xFF
+	destPtr[5] = 0xFF
+	dispatchData.copyBytes(to: destPtr, from: 9..<10)
+	expectEqual(destPtr[0], 0x15)
+	expectEqual(destPtr[1], 0xFF)
+	expectEqual(destPtr[2], 0xFF)
+	expectEqual(destPtr[3], 0xFF)
+	expectEqual(destPtr[4], 0xFF)
+	expectEqual(destPtr[5], 0xFF)
+
+	// Copy from offset 2, but only 1 byte. This copy is bounded within the 
+	// first region.
+	destPtr[1] = 0xFF
+	destPtr[2] = 0xFF
+	destPtr[3] = 0xFF
+	destPtr[4] = 0xFF
+	destPtr[5] = 0xFF
+	dispatchData.copyBytes(to: destPtr, from: 2..<3)
+	expectEqual(destPtr[0], 2)
+	expectEqual(destPtr[1], 0xFF)
+	expectEqual(destPtr[2], 0xFF)
+	expectEqual(destPtr[3], 0xFF)
+	expectEqual(destPtr[4], 0xFF)
+	expectEqual(destPtr[5], 0xFF)
+}
+
+DispatchAPI.test("DispatchData.buffers") {
+	let bytes = [UInt8(0), UInt8(1), UInt8(2), UInt8(2)]
+	var ptr = UnsafeBufferPointer<UInt8>(start: bytes, count: bytes.count)
+	var data = DispatchData(bytes: ptr)
+	expectEqual(bytes.count, data.count)
+	for i in 0..<data.count {
+		expectEqual(data[i], bytes[i])
+	}
+
+	data = DispatchData(bytesNoCopy: ptr, deallocator: .custom(nil, {}))
+	expectEqual(bytes.count, data.count)
+	for i in 0..<data.count {
+		expectEqual(data[i], bytes[i])
+	}
+
+	ptr = UnsafeBufferPointer<UInt8>(start: nil, count: 0)
+	data = DispatchData(bytes: ptr)
+	expectEqual(data.count, 0)
+
+	data = DispatchData(bytesNoCopy: ptr, deallocator: .custom(nil, {}))
+	expectEqual(data.count, 0)
+}
+
+DispatchAPI.test("DispatchData.bufferUnsafeRawBufferPointer") {
+	let bytes = [UInt8(0), UInt8(1), UInt8(2), UInt8(2)]
+	var ptr = UnsafeRawBufferPointer(start: bytes, count: bytes.count)
+	var data = DispatchData(bytes: ptr)
+	expectEqual(bytes.count, data.count)
+	for i in 0..<data.count {
+		expectEqual(data[i], bytes[i])
+	}
+
+	data = DispatchData(bytesNoCopy: ptr, deallocator: .custom(nil, {}))
+	expectEqual(bytes.count, data.count)
+	for i in 0..<data.count {
+		expectEqual(data[i], bytes[i])
+	}
+
+	ptr = UnsafeRawBufferPointer(start: nil, count: 0)
+	data = DispatchData(bytes: ptr)
+	expectEqual(data.count, 0)
+
+	data = DispatchData(bytesNoCopy: ptr, deallocator: .custom(nil, {}))
+	expectEqual(data.count, 0)
+}
+
+DispatchAPI.test("DispatchData.enumerateBytes") {
+	let count = 240
+	var bytes = [UInt8]()
+	for i in 0..<count {
+		bytes.append(UInt8(i))
+	}
+	let ptr = UnsafeRawBufferPointer(start: bytes, count: bytes.count)
+	let data = DispatchData(bytes: ptr)
+
+	var nextIndex = 0
+#if swift(>=4.2)
+	data.enumerateBytes({ ptr, offset, stop in
+		for i in 0..<ptr.count {
+			expectEqual(ptr[i + offset], UInt8(i + offset))
+		}
+		nextIndex = offset + ptr.count
+	})
+#else
+	data.enumerateBytes(block: { ptr, offset, stop in
+		for i in 0..<ptr.count {
+			expectEqual(ptr[i + offset], UInt8(i + offset))
+		}
+		nextIndex = offset + ptr.count
+	})
+#endif
```

Comment:
I suggest a trailing closure test that's expected to work in both language modes, too.

Reply:
Added that test and squashed commits.